### PR TITLE
Support CUDA 10.1 + cuDNN 7.5 + Turing

### LINKED ISSUE #2002: Support for Nvidia Tesla T4
Feature request

Current Cupy build doesn't include Tesla T4. 
https://github.com/cupy/cupy/blob/master/cupy_setup_build.py#L633. It would be great if we have support for T4 as well. I believe this architecture is more for inference than training but still a good use case to consider.
I believe it has to accommodate Tesla T4 by supporting cuda compute ability 7.5

### PR DESCRIPTION
**Pull request by kmaehashi**
Closes #2002 and #2069

Please note that:

* Error numbers of CUDA Runtime API has been renumbered in CUDA 10.1.
  https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html
* Although it is not documented in the reference, I confirmed that NVRTC from CUDA 10.0 accepts `compute_75` PTX compilation.
  https://docs.nvidia.com/cuda/archive/10.0/nvrtc/index.html#group__options
* cuDNN 7.5 changed `CUDNN_BN_MIN_EPSILON` to `0.0`.
  https://docs.nvidia.com/deeplearning/sdk/cudnn-release-notes/rel_750.html

Blocked by https://github.com/chainer/chainer-test/pull/506

### CODE REVIEW

**Comment by hvy** [comment]
@kmaehashi could you take a look at the CI failures?

**Comment by kmaehashi** [comment]
Thanks, fixed!

**Comment by kmaehashi** [comment]
Resolved conflicts.

**Comment by kmaehashi** [comment]
Test passed.

**Comment by toslunar** [comment]
chainer/chainer-test#506 is merged.

Jenkins, test this please.

File: cupy/cuda/cudnn.pyx [hunk 1]
```diff
@@ -679,6 +679,9 @@ cdef extern from 'cupy_cudnn.h' nogil:
     # Build-time version
     int CUDNN_VERSION
 
+    # Constants
+    float _CUDNN_BN_MIN_EPSILON 'CUDNN_BN_MIN_EPSILON'
```
Comment (hvy) [inline cupy/cuda/cudnn.pyx]: Should it not be a `double`?

File: cupy/cuda/cupy_cuda.h [hunk 2]
```diff
@@ -150,6 +150,8 @@ CUresult cuLaunchKernel(...) {
 
 typedef enum {
     cudaSuccess = 0,
+    cudaErrorInvalidValue = 1,
+    cudaErrorMemoryAllocation = 2,
```
Comment (hvy) [inline cupy/cuda/cupy_cuda.h]: Just an additional reference to the direct link https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1g3f51e3575c2178246db0a94a430e0038

**Comment by hvy** [comment]
Could you fix the CI failures related to the relaxation of the eps in BN?

File: cupy/cuda/cudnn.pyx (hunk 1 above)
Comment (kmaehashi) [reply cupy/cuda/cudnn.pyx]: Thanks! It should be `double` as `epsilon` of `cudnnBatchNormalizationForwardTraining` etc. are typed as `double`.

**Comment by hvy** [comment]
Blocked by #6710.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
@kmaehashi Hm, can you tell if the CI failures are related to this PR? ConvNd tests are failing in the py2 job with cudnn.

**Comment by hvy** [comment]
Discussed offline, CI failures seem unrelated. LGTM!

**Pull request merged**